custom-pages/login/src/services/AuthApi.js

```diff
@@ -29,22 +29,22 @@ export default class AuthApi {
     this.instance = new auth0.WebAuth(params)
   }
 
-  async login(username, password) {
-    return this.instance.login(
+  login(username, password, errCallback) {
+    this.instance.login(
       {
         realm: authOpts.connection,
         username,
         password
       },
       (err) => {
         if (err) {
-          if (!err.message) {
-            err.message = 'Invalid username or password'
+          if (errCallback) {
+            setTimeout(() => errCallback('Invalid username or password'), 5)
```

Comment:
is the setTimeout needed?

Reply:
as commented in another PR this is needed, React16 solved this problem with some internal hooks (this is react14)